### PR TITLE
Enhance Status Page with Application Status, Feedback, and Discord Invite Link

### LINKED ISSUE #903: Enhance /intro Page with Application Status, Feedback, and Discord Invite Link
## Issue Description
After a new user submits the join form, they receive a personalized link to check the status of their application. This link should direct them to the /intro page, formatted as `/intro?id=<user_id>&status=submitted`. The /intro page should display the application status and any feedback provided by a superuser and if the application is approved user can view discord invite link, otherwise that their application was not approved.

### Behavior for Different Users:

- ### New Users:
When a new user visits the /intro page using their personalized link, they see the status of their application. Depending on the status:

   - Pending:
     The page displays a message asking them to wait until their application is reviewed.
     `We are reviewing your application. Please check again here periodically for an update.`
     
   - Approved:
     The page provides a Discord invite link, allowing them to join the server.
     `Congratulations. You've been approved to join our Discord server using this link: <Link>`

   - Rejected:
     The page shows the rejection message and any feedback provided by the superuser.
      `Unfortunately, at this time, your application is not approved.`

     
- ### Superusers:
If a superuser accesses the personalized link, they should be redirected to the dashboard application page at https://dashboard.realdevsquad.com/applications?id=<user_id> where they can review and update the application status.

## Expected Behavior

- New users receive a personalized link after submitting the join form and can use it to check their application status on the /intro page.
- The /intro page shows different content based on the application status:
    - If the application is under review, a waiting message is displayed.
    - If the application is approved, a Discord invite link is shown.
    - If the application is rejected, a rejection message and feedback are displayed.
- Superusers are automatically redirected to the dashboard's application page (https://dashboard.realdevsquad.com/applications?id=<user_id>) when they visit the personalized link.

### Steps to Reproduce
1. A new user fills out and submits the join form.
2. The user receives a personalized link (e.g., /intro?id=<user_id>&status=submitted) and uses it to check their application status.
3. If the user is a superuser, visiting the link redirects them to https://dashboard.realdevsquad.com/applications?id=<user_id>.
4. If the user is not a superuser, they stay on the /intro page and see the relevant application status and feedback.
5. If the super user approves the application user can discord the invite link on that page.

### Severity/Priority
 - [x] Critical
 - [ ] High
 - [ ] Medium
 - [ ] Low

### Additional Information
- The Discord invite link is only provided on the /intro page when the application is approved.
- Superusers update the application status and provide feedback through the dashboard, not directly on the /intro page and the status and feedback if provided should reflect on the intro page.

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by vinit717**
Date: `19-08-2024`

Developer Name: `vinit`

---

## Issue Ticket Number:-

- closes #903 

## Description:

Add a status card and based on the application user we have shown to the user and if the application goes approved then we have to give discord invite link 

Is Under Feature Flag

- [ ] Yes
- [x] No

Database changes

- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)

- [ ] Yes
- [x] No

Is Development Tested?

- [x] Yes
- [ ] No

Tested in staging?

- [ ] Yes
- [x] No

### Add relevant Screenshot below ( e.g test coverage etc. )

Screenshot_84

Screenshot_85

Screenshot_86

### CODE REVIEW

**Review by iamitprakash — APPROVED** [review]

File: app/components/stepper.js [hunk 1]
```diff
@@ -51,8 +60,8 @@ export default class StepperComponent extends Component {
   @action startHandler() {
     if (this.login.isLoggedIn && !this.login.isLoading) {
       localStorage.setItem('id', this.login.userData.id);
-      localStorage.setItem('first_name', this.login.userData.first_name);
-      localStorage.setItem('last_name', this.login.userData.last_name);
+      localStorage.setItem('first_name', this.login.userData?.first_name);
+      localStorage.setItem('last_name', this.login.userData?.last_name);
```
Comment (prakashchoudhary07) [inline app/components/stepper.js:64]: Why is this optional?
What should happend when userData is not present?

Comment (vinit717) [reply app/components/stepper.js:64]: Can we do it like this ?


```
@action startHandler() {
  if (this.login.isLoggedIn && !this.login.isLoading) {
    if (this.login.userData) {
      localStorage.setItem('id', this.login.userData.id);
      localStorage.setItem('first_name', this.login.userData.first_name || 'Unknown');
      localStorage.setItem('last_name', this.login.userData.last_name || 'User');
      this.incrementStep();
    } else {
      alert('User data is not available. Please try logging in again.');
    }
  } else {
    alert('You must be logged in to continue');
  }
}

```

for more safety check

Comment (prakashchoudhary07) [reply app/components/stepper.js:64]: But why?

File: app/services/onboarding.js [hunk 2]
```diff
@@ -101,12 +101,22 @@ export default class OnboardingService extends Service {
 
       if (response.status === 403) {
         this.toast.error(message, 'error!', TOAST_OPTIONS);
+      } else if (response.status >= 400) {
+        this.toast.error(
+          'Something went wrong! Please try again later.',
+          'Error!',
+          TOAST_OPTIONS,
+        );
```
Comment (prakashchoudhary07) [inline app/services/onboarding.js:109]: No need message?

File: app/components/stepper.js (hunk 1 above)
Comment (vinit717) [reply app/components/stepper.js:64]: If user logged in then we have user data 
if (this.login.isLoggedIn && !this.login.isLoading) {

File: app/services/onboarding.js (hunk 2 above)
Comment (vinit717) [reply app/services/onboarding.js]: Fixed

File: app/components/join-steps/status-card.js [hunk 3]
```diff
@@ -26,7 +37,46 @@ export default class StatusCardComponent extends Component {
...
+  get status() {
+    return this.args.status || this.fetchedStatus;
+  }
+
+  get feedback() {
+    return this.args.feedback || this.fetchedFeedback;
+  }
+
+  get currentStatusDetails() {
+    return this.APPLICATION_STATUSES.find((s) => s.status === this.status);
```
Comment (prakashchoudhary07) [inline app/components/join-steps/status-card.js:55]: What is this?

Comment (vinit717) [reply app/components/join-steps/status-card.js:55]: To get user application status to display over UI

Comment (prakashchoudhary07) [reply app/components/join-steps/status-card.js:55]: okay, saw the user status constant

File: app/components/join-steps/status-card.js [hunk 4]
```diff
@@ -1,13 +1,24 @@
...
+import { TOAST_OPTIONS } from '../../constants/toast-options';
+import { ANKUSH_TWITTER } from '../../constants/urls';
 
 export default class StatusCardComponent extends Component {
   @service login;
   @service router;
+  @service onboarding;
+  @service toast;
+
+  @tracked joinLink = USER_JOINED_LINK(this.login.userData?.id);
```
Comment (prakashchoudhary07) [inline app/components/join-steps/status-card.js:16]: If undefined is passed here?
won't joinLink be invalid?

Comment (vinit717) [reply app/components/join-steps/status-card.js:16]: Yes as this is join link
  return `${APPS.HOME}/intro?id=${userId}&status=submitted`;


so either we can do something like this what your suggestion
  @action
  initializeJoinLink() {
    const userId = this.login.userData?.id;
    this.joinLink = userId ? USER_JOINED_LINK(userId) ;
  }

Comment (vinit717) [reply app/components/join-steps/status-card.js]: fixed this

**Review by prakashchoudhary07 — APPROVED** [review]

**Review by prakashchoudhary07 — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**